File: app/src/main/java/com/example/androidclient/view/LoginActivity.java
package com.example.androidclient.view;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.example.androidclient.R;
import com.example.androidclient.RetrofitClient;
import com.example.androidclient.User;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class LoginActivity extends AppCompatActivity implements View.OnClickListener{

    EditText editID, editPW;
    Button btnLogin;
    TextView failText;
    String userID, userPW;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        editID = findViewById(R.id.editID);
        editPW = findViewById(R.id.editPW);
        btnLogin = findViewById(R.id.btnLogin);
        failText = findViewById(R.id.failText);
        failText.setVisibility(View.INVISIBLE);


        btnLogin.setOnClickListener(this);

    }

    @Override
    public void onClick(View v) {

        userID = editID.getText().toString();
        userPW = editPW.getText().toString();

        if(userID.equals("") || userPW.equals("")){     //비어있으면
            failText.setText("아이디나 비밀번호는 비워둘 수 없습니다.");
            failText.setVisibility(View.VISIBLE);
            Log.d("TAG","아이디나 비밀번호는 비워둘 수 없습니다.");
        }

        Log.d("TAG", "UserID: "+ userID+" UserPW: "+userPW);

        RetrofitClient retrofitClient = new RetrofitClient();

        Call<User> call = retrofitClient.apiService.getretrofitquery(new User(userID, userPW));

        call.enqueue(new Callback<User>() {
            @Override
            public void onResponse(Call<User> call, Response<User> response) {
                if (response.isSuccessful()) {
                    Log.d("TAG", "성공 "+ response.toString());
                    failText.setVisibility(View.INVISIBLE);
                    // main으로 넘기기

                }
            }

            @Override
            public void onFailure(Call<User> call, Throwable t) {
                failText.setText("아이디나 비밀번호가 일치하지 않습니다.");
                failText.setVisibility(View.VISIBLE);
                Log.d("TAG","아이디나 비밀번호가 일치하지 않습니다.");
            }
        });
    }
}

File: app/src/main/java/com/example/androidclient/view/LoginActivity.kt
package com.example.androidclient.view

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import cn.pedant.SweetAlert.SweetAlertDialog
import com.example.androidclient.R
import com.example.androidclient.data.request.User
import com.example.androidclient.data.request.UserRequest
import com.example.androidclient.data.response.Status
import com.example.androidclient.data.response.UserResponse
import com.example.androidclient.module.SignInDialog
import com.example.androidclient.retrofit.RetrofitClient
import com.example.androidclient.room.DataBase
import com.example.androidclient.room.UserDataBase
import kotlinx.android.synthetic.main.activity_login.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response


class LoginActivity : AppCompatActivity(), View.OnClickListener {
    var editID: EditText? = null
    var editPW: EditText? = null
    var userID: String? = null
    var userPW: String? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        editID = findViewById(R.id.editID)
        editPW = findViewById(R.id.editPW)
        failText.visibility = View.INVISIBLE
        btnLogin.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        userID = editID!!.text.toString()
        userPW = editPW!!.text.toString()
        if (userID == "" || userPW == "") {     //비어있으면
            failText!!.text = "아이디나 비밀번호는 비워둘 수 없습니다."
            failText!!.visibility = View.VISIBLE
            Log.d("TAG", "아이디나 비밀번호는 비워둘 수 없습니다.")
        }

        Log.d("TAG", "UserID: $userID UserPW: $userPW")
        val retrofitClient = RetrofitClient.getInstance()
        val call =
            retrofitClient.login(
                UserRequest(
                    userID!!,
                    userPW!!
                )
            )
        call.enqueue(object : Callback<Status> {
            override fun onResponse(
                call: Call<Status>,
                response: Response<Status>
            ) {
                if (response.isSuccessful) {
                    Log.d("TAG", "성공 $response")
                    failText!!.visibility = View.INVISIBLE
                    DataBase.getInstance(applicationContext)!!.dao().delete()
                    getUserInfo(userID!!)
                } else {
                    failText!!.text = "아이디나 비밀번호가 일치하지 않습니다."
                    failText!!.visibility = View.VISIBLE
                }
            }

            override fun onFailure(
                call: Call<Status>,
                t: Throwable
            ) {
                failText!!.text = "아이디나 비밀번호가 일치하지 않습니다."
                failText!!.visibility = View.VISIBLE
                Log.d("TAG", t.toString())
            }
        })
    }

    fun getUserInfo(id : String){

        val sweetAlertDialog =
            SweetAlertDialog(this, SweetAlertDialog.PROGRESS_TYPE)
        sweetAlertDialog.progressHelper.barColor = Color.parseColor("#0DE930")
        sweetAlertDialog
            .setTitleText("로그인중")
            .setCancelable(false)
        sweetAlertDialog.show()

        val signInDialog = SignInDialog()

        RetrofitClient.getInstance().getUser(User(id)).enqueue(object : Callback<UserResponse>{
            override fun onFailure(call: Call<UserResponse>, t: Throwable) {}

            override fun onResponse(call: Call<UserResponse>, response: Response<UserResponse>) {
                signInDialog.connectionSuccess(
                    response,
                    this@LoginActivity,
                    sweetAlertDialog
                )
            }

        })
    }
}